MdeModulePkg/Bus/Pci/XhciDxe/Xhci.c

```diff
@@ -463,9 +463,10 @@ XhcGetRootHubPortStatus (
   //
   // Poll the root port status register to enable/disable corresponding device slot if there is a device attached/detached.
   // For those devices behind hub, we get its attach/detach event by hooking Get_Port_Status request at control transfer for those hub.
+  // Save the status so we can reset the port on error.
   //
   ParentRouteChart.Dword = 0;
-  XhcPollPortStatusChange (Xhc, ParentRouteChart, PortNumber, PortStatus);
+  Status                 = XhcPollPortStatusChange (Xhc, ParentRouteChart, PortNumber, PortStatus);

```

Comment:
Sorry for a question.
Do you think it is feasible to add below codes after the above line and without changing UsbBusDxe/UsbEnumer.c?
  //
  // Force resetting the port by clearing the USB_PORT_STAT_C_RESET bit in PortChangeStatus
  // when XhcPollPortStatusChange fails
  //
  if (EFI_ERROR (Status)) {
    PortStatus->PortChangeStatus &= ~(USB_PORT_STAT_C_RESET);
    Status = EFI_SUCCESS;
  }

I am not very sure the affect of doing so on the below check within UsbEnumeratePort:
  //
  // Only handle connection/enable/overcurrent/reset change.
  // Usb super speed hub may report other changes, such as warm reset change. Ignore them.
  //
  if ((PortState.PortChangeStatus & (USB_PORT_STAT_C_CONNECTION | USB_PORT_STAT_C_ENABLE | USB_PORT_STAT_C_OVERCURRENT | USB_PORT_STAT_C_RESET)) == 0) {
    return EFI_SUCCESS;
  }

Reply:
Thank you. I tried the above and it doesn't solve the issue completely. It seems the problem comes from `MdeModulePkg/Bus/Usb/UsbBusDxe/UsbEnumer.c` ln 806: `Status = UsbGetMaxPacketSize0 (Child);`

If the max packet size returned there is less than the default max size given the device speed, then it fails. i.e. You have a USB3 SS device (speed 3) which returns 64 as max size (vs 512), it will fail to get the descriptor.

Strangely, the value returned by that function seems to vary - I can get 512 one time and 64 another from same device, same firmware, etc

Would you have any suggestions?